libpod/container_log.go

```diff
@@ -81,5 +84,31 @@ func (c *Container) readFromLogFile(options *logs.LogOptions, logChannel chan *l
 		}
 		options.WaitGroup.Done()
 	}()
+	// Check if container is still running or paused
+	if options.Follow {
+		go func() {
+			for {
+				state, err := c.State()
+				time.Sleep(watch.POLL_DURATION)
```

Comment:
is there any reason why we do a `Sleep` before checking for the error?  Should the `Sleep` happen later?

Reply:
Yes. Under all conditions you must call `stopAtEOF()` when exiting the subroutine in order to break the infinite wait state `t.Lines` in the other subroutine. You must also wait up to the POLL_DURATION before calling `stopAtEOF()` or else you may run into the condition where the container has exited or does not exist and `t.Lines` is still polling for changes to the log file.

If you check commit https://github.com/containers/libpod/pull/6702/commits/8926e1f03bc82d75a295a0784c2756ea37c9b031 you will see there originally was 3 sleep statements, 2 of them after the `if` condition. This was refactored as there is no logical different between calling sleep first at the beginning versus calling it after the conditional statements and at the end of the entire loop.